server/src/internalClusterTest/java/org/elasticsearch/health/UpdateHealthInfoCacheIT.java

```diff
@@ -151,6 +128,40 @@ private DiscoveryNode waitAndGetHealthNode(InternalTestCluster internalCluster)
         return healthNode[0];
     }
 
+    /**
+     * This method fetches the health data using FetchHealthInfoCacheAction. It does this using both a random non-health-node client and
+     * a health node client. It asserts that all expected nodeIds are there with GREEN status, and that the notExpectedNodeId (if not
+     * null) is not in the results.
+     * @param internalCluster The cluster to use to get clients
+     * @param healthNode The health node
+     * @param expectedNodeIds The list of nodeIds we expect to see in the results (with a GREEN status)
+     * @param notExpectedNodeId A nullable nodeId that we do _not_ expect to see in the results
+     * @throws Exception
+     */
+    private void assertResultsCanBeFetched(
```

Comment:
To simplify this what if we pick a node randomly? Isn't this good enough?

Reply:
That's what it's doing right? It picks a random non-health node, and then also the health node (to cover the 2 main cases).

Reply:
Yes, it does both, I was wondering since as you said the routing is tested in the `TransportHealthNodeAction`, that maybe it's sufficient to just pick a node and fetch the data once. Without differentiating between the health and other nodes. WDYT?